### PR TITLE
COMP: Remove in-class {} member initializers of unique_ptr

### PR DESCRIPTION
**Pull request by SimonRit**
Errors showed up in (e.g.) https://my.cdash.org/viewBuildError.php?buildid=2273357

Patch 5e2c49f9eb80e78903ce8f9fd2b5952062653cab causes the following errors on GCC 7.4.1:
/usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘itk::NiftiImageIO::NiftiImageProxy’ /usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘itk::LBFGSBOptimizerHelper’ /usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘gdcm::SerieHelper’ /usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘itk::NiftiImageIO::NiftiImageProxy’ /usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘itk::GiftiMeshIO::GiftiImageProxy’ /usr/include/c++/7/bits/unique_ptr.h:76:22: error: invalid application of ‘sizeof’ to incomplete type ‘itk::GiftiMeshIO::GiftiImageProxy’

The issue has already been fixed in the past, see
d7c0128e99f02cfb58e8c447d175008b2b946a9f

All instances have been removed with the script:
for i in $(find Modules -type f)
do
  sed -i "s/\(unique_ptr.*\) *{}/\1/g" $i
done

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by N-Dekker** [comment]
Wow, thanks @SimonRit I knew about an in-class `{nullptr}` being problematic because of this unique_ptr bug, but I did not know that even an empty {} is still problematic for this old GCC version.

File: Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h [hunk 1]
```diff
@@ -415,9 +415,9 @@ class ITK_TEMPLATE_EXPORT BSplineInterpolateImageFunction : public InterpolateIm
   bool m_UseImageDirection{};
 
   ThreadIdType                          m_NumberOfWorkUnits{};
-  std::unique_ptr<vnl_matrix<long>[]>   m_ThreadedEvaluateIndex {};
-  std::unique_ptr<vnl_matrix<double>[]> m_ThreadedWeights {};
-  std::unique_ptr<vnl_matrix<double>[]> m_ThreadedWeightsDerivative {};
+  std::unique_ptr<vnl_matrix<long>[]>   m_ThreadedEvaluateIndex ;
```
Comment (dzenanz) [inline Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h:418]: Remove space before semicolon. linter [complains](https://github.com/InsightSoftwareConsortium/ITK/actions/runs/3968371684/jobs/6801455642)

Comment (N-Dekker) [reply Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h]: These spaces were actually introduced automatically by Clang-Format (with by PR #3851), when adding the `{}`! I think it's a Clang-Format bug!

Comment (SimonRit) [reply Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.h]: Just force pushed a slightly modified version of the regexp to cope with this, linter is happy now!

**Review by N-Dekker — APPROVED** [review]
As far as I can see, the old GCC errors only occur for `unique_ptr<T>` data members whose type `T` is a forward-declared class. So it should not affect _all_  unique_ptr data members.

Anyway, I see that it's easiest to just remove the in-class `{}` initializers from all of them. Approved.

**Pull request merged**